### PR TITLE
Initial support of GS40

### LINKED ISSUE #2667: gnome-shell 40: process changes to yaru
- [ ] wait for gnome-shell 40 to land in Ubuntu (probably 21.10)
- [x] copy+paste scss code from upstream->src
- [x] reapply the Yaru redesign

### PR DESCRIPTION
**Pull request by Jupi007**
- [x] Import scss code from upstream.
- [x] Reimport Yaru changes.
- [x] Finish light variant.
- [x] Reinsert transparent backgrounds on panel-buttons.
- [x] Fix app-folder page preview

See screenshots in comments below 👇

Closes #2667

### CODE REVIEW

**Comment by Muqtxdir** [comment]
@Jupi007 ping me whenever this is ready I'll try it out. I've successfully managed to install GNOME 40 on Ubuntu 21.04

**Comment by Jupi007** [comment]
@Muqtxdir The build CI pass, so I think you can test it now :)
Btw did you do something special to install GS 40?

**Comment by Muqtxdir** [comment]
@Jupi007 This is awesome, it works well. see the screen-record below 

https://user-images.githubusercontent.com/54065734/111883566-2e758500-89e2-11eb-8cad-e8f33591df8f.mp4

I'll do a better test tomorrow and report the issues ;)

>  Btw did you do something special to install GS 40?

I tried the PPA that I mentioned on a older-build iso of hirsute turns out `libedataserver` is causing conflicts with the .iso. Have you tried approaching the PPA guy regarding the issue with GDM?

**Comment by Feichtmeier** [comment]
Thanks @Jupi007 🚀 

Looks pretty good !
I think, if we wont have a dock in the future, we REALLY should change this hyper dark :checked state to some transparent white in the dash

**Comment by Jupi007** [comment]
> @Jupi007 This is awesome, it works well. see the screen-record below
> hirsutevm.mp4
> 
> I'll do a better test tomorrow and report the issues ;)
> 
> > Btw did you do something special to install GS 40?
> 
> I tried the PPA that I mentioned on a older-build iso of hirsute turns out `libedataserver` is causing conflicts with the .iso. Have you tried approaching the PPA guy regarding the issue with GDM?

That looks so good. I'm really looking forward to testing 😃

**Comment by Jupi007** [comment]
> I think, if we wont have a dock in the future, we REALLY should change this hyper dark :checked state to some transparent white in the dash

I agree 👍

**Comment by Jupi007** [comment]
Here is the list of the modifications I did:

- Smaller panel height
- Use our own palette for `.panel-button`
- Orange and smaller close button in overview
- Use a color between `$dash_background_color` and `$bg_color` for overview background color (`$system_bg_color`)
- Use the same background color as dash  for search entry (as the overview use a dark background color instead of the wallpaper, the old background wasn't visible enough)
- Add border to app-folders
- Lighten checked background color for overview-icons (@Feichtmeier it is that you was thinking of?)

## Screenshots

1Capture d’écran du 2021-03-21 16-35-31
Capture d’écran du 2021-03-21 16-34-46
Capture d’écran du 2021-03-21 16-35-06

**Comment by Feichtmeier** [comment]
Looks very good! @Jupi007

**Comment by chrisjbillington** [comment]
Thanks for this work!

I'm using this on Arch Linux with GS40, and I've pointed the `yaru` [AUR package](https://aur.archlinux.org/pkgbase/yaru/) at this branch as well, so others are using it too.

I haven't seen any issues, and nobody has reported any problems on the AUR page, so things seem to be pretty good!

**Comment by Jupi007** [comment]
> Thanks for this work!
> 
> I'm using this on Arch Linux with GS40, and I've pointed the `yaru` [AUR package](https://aur.archlinux.org/pkgbase/yaru/) at this branch as well, so others are using it too.
> 
> I haven't seen any issues, and nobody has reported any problems on the AUR page, so things seem to be pretty good!

Really good news 😃 Thanks @chrisjbillington

**Comment by reC0D3D** [comment]
There are a few visual inconsistencies that I've noticed while using this branch.

Screenshot from 2021-04-19 14-19-13

For starters, I feel the workspace previews under the search bar should use a dark gray (same as app folders) to match the search bar and dock. In their current state, they look out of place.

Screenshot from 2021-04-19 14-20-03

There is also a jarring inconsistency between the search bubbles and the dock. I believe the search result bubbles should match the dock.

Screenshot from 2021-04-19 14-16-26

You may need to look closely at this one. The highlight effect has a slightly smaller corner radius compared to the app folders. This leaves a few bright pixels around the corners of the folders, seen above.

**Comment by Jupi007** [comment]
Hi @reC0D3D,
thanks a lot for your review.

I fixed the app folder border-radius:

Capture d’écran du 2021-04-20 20-20-38

And also the search-entries background:

Capture d’écran du 2021-04-20 20-19-40

And finally the workspace previews:

Capture d’écran du 2021-04-20 19-26-31

**Comment by reC0D3D** [comment]
This is marvelous, @Jupi007! Great work

**Comment by reC0D3D** [comment]
It might be worth exploring the design for the light shell theme as the dark one seems spot-on. 

- What about experimenting with a light colored overview background? (Same color as header bar on Yaru-light.) Although...I'm sure users may complain about brightness.
- Alternatively, keep the dark background but lighten certain elements? Those could include the folders, dash, search bar, and previews. This would be similar to the way Yaru-light on 3.38 is handled. However, drastic shifts in contrast in this application is not elegant in my opinion.
- Maybe the overview can continue to be dark? It would definitely clash but it is a solution.

I would like to hear some other opinions on this topic as I _surprisingly_ have a bias towards light themes. (I do not mind bright whites and blinding my eyes)

**Comment by Feichtmeier** [comment]
Not a bad idea, but this would require to make this in upstream, and while we worked upstream to create a light shell, the GNOME designers explicitly wanted to keep some parts, like the panel and the OSD, dark, even in the light theme.

Since they still didn't figure out yet how to sync the shell theme with the gtk theme (without user themes, which they say is unsafe) they will probably stick to the full dark shell for a while. We just had the chance to create a light shell because they default to this in the GNOME classic session used by redhat enterprise Linux.

So long story short: we stick to upstream with light/dark theming

**Comment by reC0D3D** [comment]
Gotcha, that makes sense. Thanks for the info, @Feichtmeier!

**Comment by plumlis** [comment]
opIt seems yaru theme gnome 40 branch doesn't works with opacity option on Dash to dock extensions(Or maybe Ubuntu dock?)

**Comment by chrisjbillington** [comment]
> It seems yaru theme gnome 40 branch doesn't works with opacity option on Dash to dock extensions(Or maybe Ubuntu dock?)

Dash to panel's gnome 40 branch is having difficulties getting transparency working regardless of theme, so this seems likely to be an issue of then panel/dock extensions and how they interact with the new shell rather than the themes.

**Comment by Feichtmeier** [comment]
Several iterations will happen upstream and it is also not clear what is happening to the ubuntu dock, but expect yaru to be ready for 21.10, please don't panic :)

**Comment by Jupi007** [comment]
> This issue only affects a tiny percentage of users, but it may be something to consider. The selection bubble in the panel uses a solid color instead of a transparent color like upstream Adwaita. In normal use, this is perfectly fine. Although, a select few do enjoy modifying the panel via extensions. As seen in the screenshot below, the select effect is jarring with a transparent panel.
> 
> Screenshot from 2021-05-04 10-30-42

Thanks for the report, it is on the todo list. 👍

**Comment by reC0D3D** [comment]
I noticed that these previews (on the right of the folder) still look like Adwaita. For example, the arrow is a bit chunky compared to Yaru and its icon set. The corner radius is also a bit round compared to the rest of the UI.

Screenshot from 2021-05-07 15-03-34

**Comment by Jupi007** [comment]
> I noticed that these previews (on the right of the folder) still look like Adwaita. For example, the arrow is a bit chunky compared to Yaru and its icon set. The corner radius is also a bit round compared to the rest of the UI.

Thanks @reC0D3D, I'll investigate.

**Comment by reC0D3D** [comment]
The page previews now look great, @Jupi007! However, it appears that the close button icon in the overview broke somewhere in those last three commits. 

Screenshot from 2021-05-09 15-04-20

**Comment by Muqtxdir** [comment]
@reC0D3D I just pushed `window-close-24-symbolic.svg` for the close-button in overview

**Comment by reC0D3D** [comment]
I figured out the reason why my close button disappeared — it was unrelated to that commit. Sorry for the confused report! Thanks for it, regardless, @Muqtxdir! It should help out someone.

I experimented by setting the `gnome-shell-gresource` option in `meson_options.txt` to true. I tried this because I noticed that the master branch set it to true it by default, while this one doesn't. Maybe it would still be worth looking into?

**Comment by Jupi007** [comment]
Thanks a lot for the fix @Muqtxdir !

@reC0D3D I merged the master branch, so `gnome-shell-gresource` option is always true now.

**Comment by chrisjbillington** [comment]
I'm seeing two issues here on Arch Linux.

One is window buttons on windows with client-side decorations: 

image

The other is that the dark shell theme is no longer showing up in Gnome tweaks, though Yaru-light still is. This confuses me, since both `/usr/share/themes/Yaru{,-light}/gnome-shell` exist on my system.

Could be an issue with how the package is built on gnome-shell - the build is just this:

```bash
build() {
    arch-meson yaru build
    meson configure build -Dubuntu-unity=true
    ninja -C build
}
```

where `arch-meson` is the wrapper script:
```bash
#!/bin/bash -ex
# Highly opinionated wrapper for Arch Linux packaging

exec meson setup \
  --prefix        /usr \
  --libexecdir    lib \
  --sbindir       bin \
  --buildtype     plain \
  --auto-features enabled \
  --wrap-mode     nodownload \
  -D              b_lto=true \
  -D              b_pie=true \
  "$@"

```

**Comment by LinuxLlama** [comment]
Huh, unusual — I misunderstood your issue. I'm not having this issue on my end but I'll see if I can reproduce it!

**Comment by guihkx** [comment]
@chrisjbillington Building with `-Dgnome-shell-gresource=false` fixed it for me, i.e. GNOME Tweaks now displays `Yaru` as a Shell theme.

Though as I don't know what this "gresource" thing is, I'm not sure if there are any implications of disabling it.

**Comment by chrisjbillington** [comment]
Thanks @guihkx , that's great. I admit I don't understand why it's failing when the gresource option is enabled (the shell themes in /usr/share/themes/, where the user themes extension looks, were not packed into a gresource file regardless of the option being set, only the one in /usr/share/gnome-shell/theme), but I'll just set the option false, revert the change to the PKGBUILD and scratch my head about the window buttons issue instead.

**Comment by chrisjbillington** [comment]
Verified that the window buttons issue depends on the icon theme. Even if the GTK theme is Adwaita, the buttons are still showing the wrong icons depending on whether the Yaru icon theme is selected or not.

Peek 2021-05-12 16-42

**Comment by Jupi007** [comment]
Thanks for your PR @chrisjbillington. Me and @Muqtxdir have updated Yaru's symbolic, but I didn't know that it was also needed to update `index.theme` to reflect new icon folder.

Also, I made a PR which revert the `gresource` option to false: #2832

**Comment by kenvandine** [comment]
@3v1n0 Can you please help review this?

**Comment by Jupi007** [comment]
Just a note: this is still WIP. I'm currently working on the missing transparent areas.
See the todo list of first comment 👆

**Comment by guihkx** [comment]
No worries, thanks for the heads up too! Love the work you guys did with this theme. 🤗

**Comment by Jupi007** [comment]
I finished the light variant (was just missing to fix dark elements border-color).

Capture d’écran de 2021-05-13 20-09-55

Just a note: I used the same look as upstream (dark overview background). If we want a light background we need to discuss that (personally I like the current look).

**Comment by 3v1n0** [comment]
I'll give this a look, if @Feichtmeier is fine we can probably already merge this early so that we can start having it built for impish and update it new iterations, ok?

**Comment by Feichtmeier** [comment]
Okay, would still be great if you can keep an eye on this, since I am currently not able to schedule as much time on yaru gnome themes as previously.

**Comment by Jupi007** [comment]
Actually, this branch is totally ready to use (I'm daily use it). All problems are fixed.
So should I merge this PR? 

@Feichtmeier @3v1n0

**Comment by Feichtmeier** [comment]
Okay, eventually mark the readme that the gnome shell theme from master now needs Gnome 40.

**Comment by Jupi007** [comment]
> Okay, eventually mark the readme that the gnome shell theme from master now needs Gnome 40.

Good idea. But should I indicate that in the README or into CONTRIBUTING (which contain install instructions)?

**Comment by Feichtmeier** [comment]
I think in the read me. There is already a warning to not install from source for daily usage, maybe in this region
